node/src/types/block/meta_block/state.rs

```diff
@@ -50,23 +49,39 @@ impl State {
         Self::default()
     }
 
-    /// Returns a new `State` with all fields set to `false` except for
-    /// `immediate_switch_block_for_current_protocol_version`.
-    pub(crate) fn new_immediate_switch() -> Self {
+    /// Returns a new `State` with all fields set to `false` except for `gossiped`.
+    pub(crate) fn new_not_to_be_gossiped() -> Self {
         State {
-            immediate_switch_block_for_current_protocol_version: true,
+            gossiped: true,
             ..Self::default()
         }
     }
 
     /// Returns a new `State` with all fields set to `false` except for `stored`.
-    pub(crate) fn new_synced() -> Self {
+    pub(crate) fn new_already_stored() -> Self {
         State {
             stored: true,
             ..Self::default()
         }
     }
 
+    /// Returns a new `State` with all fields set to `true` except for `sent_to_deploy_buffer` and
+    /// `sent_to_consensus_post_execution`.
```

Comment:
Nit:

```suggestion
    /// `consensus_notified`.
```

but, tbh, I'd rather remove the comment or rephrase it to something like "returns a new State which the block is expected to have after it has been synced"...

Reply:
Done in [8eeb2f8](https://github.com/casper-network/casper-node/pull/3569/commits/8eeb2f8ba77159331b3a25a1bbfc423618df8f60).